subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/worker/device/adb/AdbDevice.kt

```diff
@@ -410,11 +425,23 @@ data class AdbDevice(
         )
 
     override fun log(message: String) {
-        logger.log("[$id $api] $message")
+        logger.log("$TAG $message")
     }
 
     override fun notifyError(message: String, error: Throwable) {
-        logger.notify("[$id $api] $message", error)
+        logger.notify("$TAG $message", error)
+    }
+
+    override fun toString(): String {
+        return "Device $TAG"
+    }
+
+    private val TAG: String by lazy {
+        try {
+            "[$id $api]"
+        } catch (ignore: Exception) {
```

Comment:
It looks crazy that we catch all exceptions only for retrieving tag value. When does exception can occur here?

Reply:
This is a hypothetical case when a device is frozen.
We don't want to lose an error message in this case.

We know how to avoid this problem. ReservationClient can send validated Device with already checked API level. Hope to do it in MBS-8531

Reply:
Thanks for pointing attention. It was a recursion here.
Removed this logic to avoid such problems.